datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -627,7 +627,7 @@ def should_bypass_proxy(url, no_proxy_uris):
 
 def create_basic_auth(config):
     # Since this is the default case, only activate when all fields are explicitly set
-    if config['username'] and config['password']:
+    if config['username'] is not None and config['password'] is not None:
```

Comment:
Now that you're on it should we switch to using getters, that are safer than direct access?
```suggestion
    if config.get('username') is not None and config.get('password') is not None:
```